tests/ui/parser/ternary_operator.stderr

```diff
@@ -0,0 +1,113 @@
+error: Rust has no ternary operator
+  --> $DIR/ternary_operator.rs:4:19
+   |
+LL |     let x = 5 > 2 ? true : false;
+   |                   ^^^^^^^^^^^^^^^
+   |
+   = help: use an `if-else` expression instead
+
+error: Rust has no ternary operator
+  --> $DIR/ternary_operator.rs:13:19
+   |
+LL |     let x = 5 > 2 ? { true } : { false };
+   |                   ^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: use an `if-else` expression instead
+
+error: Rust has no ternary operator
+  --> $DIR/ternary_operator.rs:22:19
+   |
+LL |     let x = 5 > 2 ? f32::MAX : f32::MIN;
+   |                   ^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: use an `if-else` expression instead
+
+error: expected one of `.`, `;`, `?`, `else`, or an operator, found `:`
+  --> $DIR/ternary_operator.rs:31:37
+   |
+LL |     let x = 5 > 2 ? { let x = vec![]: Vec<u16>; x } : { false };
+   |                                     ^ expected one of `.`, `;`, `?`, `else`, or an operator
+
+error: Rust has no ternary operator
+  --> $DIR/ternary_operator.rs:31:19
+   |
+LL |     let x = 5 > 2 ? { let x = vec![]: Vec<u16>; x } : { false };
+   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = help: use an `if-else` expression instead
+
+error[E0277]: the `?` operator can only be applied to values that implement `Try`
+  --> $DIR/ternary_operator.rs:4:17
+   |
+LL |     let x = 5 > 2 ? true : false;
+   |                 ^^^ the `?` operator cannot be applied to type `{integer}`
+   |
+   = help: the trait `Try` is not implemented for `{integer}`
```

Comment:
Hmm... it's somewhat disappointing that this is still being parsed as a try expr... :thinking:

Reply:
Something like:

```
if has_question {
    // `expr?`
    e = self.mk_expr(lo.to(self.prev_token.span), ExprKind::Try(e));
    if self.may_recover() && self.token == tok::Colon {
        // try recover here, setting up `e` to be an if statement so that typeck is happy...
    }
    continue;
}
```

Reply:
Yeah ideally it wouldn't, that part of it still feels very terse. I don't see where `?` is parsed though, can you give some pointers? I was thinking it could perhaps then modify the previous expression to remove the try.

Reply:
https://github.com/rust-lang/rust/blob/1821920cc8e7dc44b0035da890170bf3eadc2ae9/compiler/rustc_parse/src/parser/expr.rs#L962-L975

Reply:
But to be very clear, I'd be happy to approve this without having to rewrite this another time, if you address all the other comments above.

Reply:
I'd gladly do it in hopes of a better diagnostic :)